File: back/src/main/java/com/superlucky/model/Subscription.java
package com.superlucky.model;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import lombok.Data;

@Entity
@Data
public class Subscription {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    @ManyToOne
    public Campaign campaign;
    @OneToOne
    public Contact contact;
    public Date subscribedAt;
    public String luckyCode;
}

File: back/src/Entity/Subscription.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";
import Campaign from "./Campaign";
import Contact from "./Contacts";

@Entity()
export default class Subscription {
    constructor() {
        this.name = "";
    }
    @PrimaryGeneratedColumn()
    id?: number;

    @ManyToMany((type) => Campaign, { cascade: false })
    @JoinTable()
    campaign?: Campaign[];

    @ManyToMany((type) => Contact, { cascade: false })
    @JoinTable()
    contact?: Contact[];

    @Column()
    subscribedAt: Date

    @Column()
    luckycode: string
}
